src/main/cli/rpc.ts

```diff
@@ -0,0 +1,374 @@
+      const results = (responses as RawRpcResponseSuccess<T>[]).map(
+        (r) => r.result
+      );
+      resolve(results);
+    } catch (e) {
+      reject(e);
+    } finally {
+      store.dispatch(rpcSetEndpointFree(method));
+      console.timeEnd(`makeRpcRequest ${method} (batch)`);
+    }
+  });
+}
+
+function providerFromGetSellerResponse(
+  providerResponse: SwapSellerInfo
+): Provider {
+  const multiAddr = new Multiaddr(providerResponse.addresses[0])
+    .decapsulate('p2p')
+    .toString();
+
+  return {
+    peerId: providerResponse.peerId,
+    multiAddr,
+    testnet: isTestnet(),
+  };
+}
+
+export async function checkBitcoinBalance() {
+  const response = await makeRpcRequest<BalanceBitcoinResponse>(
+    RpcMethod.GET_BTC_BALANCE,
+    {}
+  );
+  store.dispatch(rpcSetBalance(response.balance));
+}
+
+export async function withdrawAllBitcoin(address: string) {
+  store.dispatch(rpcResetWithdrawTxId());
+  const response = await makeRpcRequest<WithdrawBitcoinResponse>(
+    RpcMethod.WITHDRAW_BTC,
+    {
+      address,
+    }
+  );
+  store.dispatch(rpcSetWithdrawTxId(response.txid));
+}
+
+export async function getSwapInfo(swapId: string) {
+  return makeRpcRequest<GetSwapInfoResponse>(RpcMethod.GET_SWAP_INFO, {
+    swap_id: swapId,
+  });
+}
+
+export async function getSwapInfoBatch(
+  swapIds: string[]
+): Promise<GetSwapInfoResponse[]> {
+  return makeBatchRpcRequest<GetSwapInfoResponse>(
+    RpcMethod.GET_SWAP_INFO,
+    swapIds.map((swapId) => ({
+      swap_id: swapId,
+    }))
+  );
+}
+
+export async function buyXmr(
+  redeemAddress: string,
+  refundAddress: string,
+  provider: Provider
+) {
+  const { swapId } = await makeRpcRequest<BuyXmrResponse>(
+    RpcMethod.BUY_XMR,
+    {
+      bitcoin_change_address: refundAddress,
+      monero_receive_address: redeemAddress,
+      seller: providerToConcatenatedMultiAddr(provider),
+    },
+    (logs) => {
+      store.dispatch(
+        swapAddLog({
+          logs,
+          isFromRestore: false,
+        })
+      );
+    },
+    true
+  );
+
+  store.dispatch(
+    swapInitiate({
+      provider,
+      spawnType: SwapSpawnType.INIT,
+      swapId,
+    })
+  );
+}
+
+export async function cancelRefundSwap(swapId: string) {
+  const swapInfo = await getSwapInfo(swapId);
+  const previousLogs = await getSavedLogsOfSwapId(swapId);
+
+  store.dispatch(
+    swapInitiate({
+      provider: providerFromGetSellerResponse(swapInfo.seller),
+      spawnType: SwapSpawnType.CANCEL_REFUND,
+      swapId,
+    })
+  );
+
+  store.dispatch(
+    swapAddLog({
+      logs: previousLogs,
+      isFromRestore: true,
+    })
+  );
+
+  await makeRpcRequest(
+    RpcMethod.CANCEL_REFUND_SWAP,
+    {
+      swap_id: swapId,
+    },
+    (logs) => {
+      store.dispatch(
+        swapAddLog({
+          logs,
+          isFromRestore: false,
+        })
+      );
+    },
+    true
+  );
+}
```

Comment:
**Suggestion:** Consider adding error handling for the async function 'cancelRefundSwap' at line 247. Currently, if an error occurs during the execution of 'getSwapInfo', 'getSavedLogsOfSwapId' or 'makeRpcRequest', it will not be caught and could lead to unexpected behavior.
```suggestion
export async function cancelRefundSwap(swapId: string) {
  try {
    const swapInfo = await getSwapInfo(swapId);
    const previousLogs = await getSavedLogsOfSwapId(swapId);
    ...
  } catch (e) {
    logger.error(`Error occurred: ${e.message}`);
  }
}
```